docs/usage/parallelization.rst

```diff
@@ -17,120 +17,142 @@ inference via different clients. Currently ELFI includes three clients:
+giving the name of the client to ``elfi.set_client``.
 
-This tutorial shows how to activate and use the ``ipyparallel`` client
-with ELFI. For local parallelization, the ``multiprocessing`` client is
-simpler to use.
+This tutorial shows how to activate and use the ``multiprocessing`` or
+``ipyparallel`` client with ELFI. The ``ipyparallel`` client supports
+parallelization from local computer up to a cluster environment. For
+local parallelization however, the ``multiprocessing`` client is simpler
+to use. Let's begin by importing ELFI and our example MA2 model from the
+tutorial.
 
-Activating parallelization
---------------------------
+.. code:: python
 
-To activate the ``ipyparallel`` client in ELFI you just need to import
-it:
+    import elfi
+    from elfi.examples import ma2
 
-.. code:: ipython3
+Let's get the model and plot it (requires graphviz)
 
-    import elfi
-    # This activates the parallelization with ipyparallel
-    import elfi.clients.ipyparallel
+.. code:: python
 
-Starting a local ipcluster
---------------------------
+    model = ma2.get_model()
+    elfi.draw(model)
 
-Before you can actually run things in parallel you also need to start an
-``ipyparallel`` cluster. Below is an example of how to start a local
-cluster to the background using 4 CPU cores:
 
-.. code:: ipython3
 
-    !ipcluster start -n 4 --daemon
-    
-    # This is here just to ensure that ipcluster has enough time to start properly before continuing
-    import time
-    time.sleep(10)
 
-.. note:: The exclamation mark above is a Jupyter syntax for executing shell commands. You can run the same command in your terminal without the exclamation mark.
+.. image:: http://research.cs.aalto.fi/pml/software/elfi/docs/0.6.1/usage/parallelization_files/parallelization_5_0.svg
 
-.. tip:: Please see the [ipyparallel documentation](https://ipyparallel.readthedocs.io/en/latest/intro.html#getting-started) for more information and details for setting up and using ipyparallel clusters.
 
-Running parallel inference
---------------------------
 
-We will run parallel inference for the MA2 model introduced in the basic
-tutorial. A ready made model can be imported from ``elfi.examples``:
+Multiprocessing client
+----------------------
 
-.. code:: ipython3
+The multiprocessing client allows you to easily use the cores available
+in your computer. You can activate it simply by
 
-    from elfi.examples import ma2
-    model = ma2.get_model()
-    
-    elfi.draw(model)
+.. code:: python
 
+    elfi.set_client('multiprocessing')
 
+Any inference instance created after you have set the new client will
+automatically use it to perform the computations. Let's try it to our
+MA2 example model from the tutorial.
 
+When running the next command, take a look at the system monitor of your
+operating system; it should show 4 (or whatever number you gave the
+ipcluster start command) Python processes doing heavy computation
```

Comment:
I thought `ipcluster` is related to `ipyparallel` is it also needed with the `multiprocessing` cliient?